src/lang.rs

```diff
@@ -847,6 +863,15 @@ impl Context for CallStack {
         result
     }
 
+    #[inline]
+    fn eval_mutable(&mut self, expr: Expr) -> EvalResult {
+        match expr {
+            Expr::Symbol(x) => self.get_mutable(x),
+            // can happen when user does f() = 3, maybe this should throw?
+            _ => self.eval(expr),
```

Comment:
Similar thing here - how would you feel about converting this to an error?

Just in terms of maintenance, I've found it easier to track down incomplete implementations when they're explicit about throwing errors or being unimplemented.

Reply:
Great idea, I agree!. 

On a side note: I think in principle this also applies to `Obj::lazy_copy()` and `Obj::mutable_view()`, but if we throw an error there I think some tests start to break so I think there we need to live with this for now